### ISSUE TITLE
Are ++ and -- operators statements or expressions?

### ISSUE THREAD

**Issue by adeftu** [hhvm/user-documentation#1151]
# Where is the problem?

In the section [Coalescing assignment operator](https://docs.hhvm.com/hack/expressions-and-operators/coalesce#coalescing-assignment-operator):
```
$arr[++$i] ??= 42;              // $i is incremented once
$arr[++$i] = $arr[++$i] ?? 42;  // $i is incremented twice
```

# What is the problem?

According to [Expressions And Operators: Incrementing And Decrementing](https://docs.hhvm.com/hack/expressions-and-operators/incrementing-and-decrementing) section:
> Note that ++ and -- are statements, not expressions.

If the incrementing/decrementing operators are statements, then how can the code above be valid? The documentation seems to contradict itself.

--------------------------------

Please don't change anything below this point.

--------------------------------

 - Build ID: HHVM=HHVM-4.140.1:HSL=v4.108.1:2022-01-14T20:09:46+0000:1fa47f258c6b68f8ec01899aa82fd6ffa0957109
 - Page requested: /hack/expressions-and-operators/coalesce
 - Page requested at: Thu, 20 Jan 2022 01:13:24 +0000
 - Controller: GuidePageController

**Comment by AndrewDiMola** [comment]
I believe you're right in that the sample would be a parsing error.

cc: @fredemmott or @Wilfred to confirm.

**Comment by fredemmott** [comment]
Yep, ++/-- as expressions is no longer supported; it works at runtime, but that may change and should be considered invalid.
